### PR TITLE
Server side rendering - Initial PR

### PR DESCRIPTION
**Pull request by jaydhulia**
- [x] Serve a static folder
- [x] Error page
- [x] Include Auto-reload on changes to HTML
- [x] Handler for scenario UI
- [x] Handler for static files
- [x] Sample scene page

### CODE REVIEW

**Comment by jaydhulia** [comment]
image
Just prints out the information for the scene and scenario for a valid scenario.

**Comment by jaydhulia** [comment]
image

Invalid Scenario/Scene is a generic error for now (designers will update it :) )

File: app/__main__.py
Comment (ahmedhamodi) [inline app/__main__.py:60]: Is this using our local path? When this is deployed on a server, what is this going to be?

File: app/routes/ui/scenario.py
Comment (ahmedhamodi) [inline app/routes/ui/scenario.py:25]: Order of checks? Should we check invalid before is published?

Pros: if they see it's not accessible, they'll assume it exists even though it might not.

Cons: malicious users can go through and find which scene ids exist, published or not.

I'll let you decide which is best - won't block the PR for this.

**Review by ahmedhamodi — APPROVED** [review]
lgtm. Couple comments, I'll let you make the final decision on these.

File: app/routes/ui/scenario.py
Comment (jaydhulia) [reply app/routes/ui/scenario.py:25]: No, so this order is what the most secure way. If a scenario is not published, then we should leak *no* information, hence why the scene ids check is only after ensuring that the scenario is supposed to be accessible.

File: app/__main__.py
Comment (jaydhulia) [reply app/__main__.py:60]: This is the path of the file where it this file exists (__main__.py). So I am saying get the directory where __main__.py exists, and append `public/static` to it. Meaning it will always be correct, regardless of running locally or on server

**Pull request merged**
